core/include/gnuradio-4.0/Block.hpp

```diff
@@ -783,18 +843,18 @@ struct Block : protected std::tuple<Arguments...> {
                         };
                         (consume_port(input_port), ...);
                     },
-                    inputPorts(&self));
+                    inputPorts<traits::port::kind::Stream>(&self));
```

Comment:
I like the trait concept.... the `traits::` and `::kind::` afterwards is a bit redundant.

What about naming this: `traits::port::Stream` (my preference) or `port::kind::Stream`

Reply:
I've added ANY to PortType like we have for Direction, and removed `kind`